### PR TITLE
DOC-530 Azure - BYO VNET Phase 1

### PR DESCRIPTION
**Pull request by micheleRP**
## Description

Resolves https://redpandadata.atlassian.net/browse/DOC-530
Review deadline: 

Status: This is in private beta.

## Page previews
[Create a Customer-Managed VNet on Azure](https://deploy-preview-87--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/azure/vnet-azure/)

## Checks

- [X] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/get-started/pages/cluster-types/byoc/azure/vnet-azure.adoc [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+```
+rpk cloud byoc azure apply --redpanda-id='$rp_id' --subscription-id='$subscription_id'
+```
+
+== Check cluster status
+
+Go to the Repanda Cloud UI to check the progress of the cluster deployment.
+
+**Reviewers: or Cloud API?**
```
Comment (paulzhang97) [inline modules/get-started/pages/cluster-types/byoc/azure/vnet-azure.adoc:275]: Customer can use Cloud API as well. What is described for AWS or GCP BYO VPC? We should use the same for Azure to be consistent.

File: modules/get-started/pages/cluster-types/byoc/azure/vnet-azure.adoc [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+== Create a cluster with the Cloud API
+
+. A network ID is required to create a Redpanda cluster. Replace the variables with your actual values. To create a resource group and bearer token, follow the xref:manage:api/cloud-api-quickstart.adoc#try-the-cloud-api[Cloud API Quickstart].
++
+```
+cluster_post_body=`cat << EOF
+{
+  "cloud_provider": "CLOUD_PROVIDER_AZURE",
+  "connection_type": "CONNECTION_TYPE_PUBLIC",
```
Comment (paulzhang97) [inline modules/get-started/pages/cluster-types/byoc/azure/vnet-azure.adoc:210]: Change `CONNECTION_TYPE_PUBLIC` to `CONNECTION_TYPE_PRIVATE`. I will fix it in cloud-example.

File: modules/get-started/pages/cluster-types/byoc/azure/create-byoc-cluster-azure.adoc [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
 include::shared:partial$feature-flag.adoc[]
 
+NOTE: With standard BYOC clusters, Redpanda manages security policies and resources for your virtual network (VNet), including subnetworks, service accounts, IAM roles, firewall rules, and storage buckets. For the most security, you can manage these resources yourself with a xref:get-started:cluster-types/byoc/azure/vnet-azure.adoc[customer-managed VNet on Azure].
```
Comment (paulzhang97) [inline modules/get-started/pages/cluster-types/byoc/azure/create-byoc-cluster-azure.adoc:7]: Change `including subnetworks, service accounts, IAM roles, firewall rules, and storage buckets` to `including subnetworks, managed identities, IAM roles, security groups, and storage accounts`.

File: modules/get-started/pages/cluster-types/byoc/azure/vnet-azure.adoc (hunk 1 above)
Comment (micheleRP) [reply modules/get-started/pages/cluster-types/byoc/azure/vnet-azure.adoc:275]: We don't have this step in GCP. For AWS, we have [this](https://deploy-preview-12--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/aws/vpc-byo-aws/#check-cluster-status).

Comment (micheleRP) [reply modules/get-started/pages/cluster-types/byoc/azure/vnet-azure.adoc:275]: For now, I'll put: "You can check the cluster status with the Cloud API or the Redpanda Cloud UI."

Comment (paulzhang97) [reply modules/get-started/pages/cluster-types/byoc/azure/vnet-azure.adoc]: Let's use the same including the example `curl` commands for Azure.

**Review by paulzhang97 — APPROVED** [review]

**Comment by micheleRP** [comment]
Add to What's New for release

**Comment by Deflaimun** [comment]
Closed by accident. Reopening

**Comment by Deflaimun** [comment]
Moving this PR into https://github.com/redpanda-data/cloud-docs/pull/121

**Pull request closed without merging**